crates/store/src/db/sql/utils.rs

```diff
@@ -0,0 +1,120 @@
+use miden_node_proto::domain::accounts::{AccountInfo, AccountSummary};
+use miden_objects::{
+    accounts::{Account, AccountDelta},
+    crypto::hash::rpo::RpoDigest,
+    notes::Nullifier,
+    utils::Deserializable,
+};
+use rusqlite::{
+    params,
+    types::{Value, ValueRef},
+    Connection, OptionalExtension,
+};
+
+use crate::errors::DatabaseError;
+
+/// Returns the high 16 bits of the provided nullifier.
+pub fn get_nullifier_prefix(nullifier: &Nullifier) -> u32 {
+    (nullifier.most_significant_felt().as_int() >> 48) as u32
+}
+
+/// Checks if a table exists in the database.
+pub fn table_exists(conn: &Connection, table_name: &str) -> rusqlite::Result<bool> {
+    Ok(conn
+        .query_row(
+            "SELECT 1 FROM sqlite_master WHERE type = 'table' AND name = $1",
+            params![table_name],
+            |_| Ok(()),
+        )
+        .optional()?
+        .is_some())
+}
+
+/// Returns the schema version of the database.
+pub fn schema_version(conn: &Connection) -> rusqlite::Result<usize> {
+    conn.query_row("SELECT * FROM pragma_schema_version", [], |row| row.get(0))
+}
+
+/// Generates a simple insert SQL statement with parameters for the provided table and fields
+pub fn insert_sql(table: &str, fields: &[&str]) -> String {
+    format!(
+        "INSERT INTO {table} ({}) VALUES ({})",
+        fields.join(", "),
+        "?, ".repeat(fields.len()).trim_end_matches(", ")
+    )
+}
```

Comment:
Final thought: could we make this const/static somehow? This will allocate each time despite the sql being a constant str per query.

Probably requires a `macro_rules!` though. Can definitely leave that sort of thing to the benchmarking time though.

Reply:
This is nice idea, thank you! But I have no idea, how to implement it on stable Rust. I have only solution for unstable: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=4b7136a128a7e57e44c5af8c10072dbd

Reply:
If you're up for a challenge: [tt-muncher macro](https://veykril.github.io/tlborm/decl-macros/patterns/tt-muncher.html) would be one of achieving it (I believe). Its a fair bit of complication though; so likely simply writing out the sql by hand would be better all round 😆

Reply:
Thank you! Finally I've found a solution that works. Please, take a look. If it's overkilling, we still may revert back to hand-written SQLs. :)